server/src/test/java/keywhiz/commands/BackfillOwnershipCommandTest.java

```diff
@@ -0,0 +1,119 @@
+package keywhiz.commands;
+
+import static keywhiz.jooq.tables.Accessgrants.ACCESSGRANTS;
+import static keywhiz.jooq.tables.Groups.GROUPS;
+import static keywhiz.jooq.tables.Secrets.SECRETS;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+import java.time.Instant;
+import java.util.Collections;
+import java.util.Random;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+import javax.inject.Inject;
+import keywhiz.KeywhizTestRunner;
+import keywhiz.test.ServiceContext;
+import net.sourceforge.argparse4j.inf.Namespace;
+import org.jooq.DSLContext;
+import org.jooq.Record1;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+@RunWith(KeywhizTestRunner.class)
+public class BackfillOwnershipCommandTest {
+  @Inject
+  private DSLContext jooq;
+
+  @Test
+  public void backfillsOldestOwner() {
+    long secretId = insertSecret();
+
+    long group1Id = insertGroup();
+    long group2Id = insertGroup();
+    long group3Id = insertGroup();
+
+    grantAccess(secretId, group1Id);
+    grantAccess(secretId, group2Id);
+    grantAccess(secretId, group3Id);
+
+    backfill();
+
+    assertOwnerEquals(secretId, group1Id);
+  }
```

Comment:
This is good for some basic coverage, but it doesn't really handle cases like secrets which already have owners, secrets with no access grants, or multiple batches. I'll add a couple more tests to handle those.

Reply:
Actually yeah, good point, I'm going to revoke my stamp until you add in those tests :P

Reply:
I've added additional tests, and also added an API for the backfill that returns the number of batches processed so that we can verify that that's working as intended.